tests/app/service/test_statistics.py

```diff
@@ -24,44 +40,48 @@
     {
         'empty': ([], [0, 0, 0], [0, 0, 0], [0, 0, 0]),
         'always_increment_requested': (
-            [StatsRow('email', 'delivered', 1), StatsRow('email', 'failed', 1)],
+            [StatsRow(EMAIL_TYPE, NOTIFICATION_DELIVERED, 1), StatsRow(EMAIL_TYPE, NOTIFICATION_FAILED, 1)],
             [2, 1, 1],
             [0, 0, 0],
             [0, 0, 0],
         ),
         'dont_mix_template_types': (
-            [StatsRow('email', 'delivered', 1), StatsRow('sms', 'delivered', 1), StatsRow('letter', 'delivered', 1)],
+            [
+                StatsRow(EMAIL_TYPE, NOTIFICATION_DELIVERED, 1),
+                StatsRow(SMS_TYPE, NOTIFICATION_DELIVERED, 1),
+                StatsRow(LETTER_TYPE, NOTIFICATION_DELIVERED, 1),
+            ],
             [1, 1, 0],
             [1, 1, 0],
             [1, 1, 0],
         ),
         'convert_fail_statuses_to_failed': (
             [
-                StatsRow('email', 'failed', 1),
-                StatsRow('email', 'technical-failure', 1),
-                StatsRow('email', 'temporary-failure', 1),
-                StatsRow('email', 'permanent-failure', 1),
-                StatsRow('letter', 'validation-failed', 1),
-                StatsRow('letter', 'virus-scan-failed', 1),
-                StatsRow('letter', 'permanent-failure', 1),
-                StatsRow('letter', 'cancelled', 1),
+                StatsRow(EMAIL_TYPE, NOTIFICATION_FAILED, 1),
+                StatsRow(EMAIL_TYPE, NOTIFICATION_PERMANENT_FAILURE, 1),
+                StatsRow(EMAIL_TYPE, NOTIFICATION_TEMPORARY_FAILURE, 1),
+                StatsRow(EMAIL_TYPE, NOTIFICATION_PERMANENT_FAILURE, 1),
```

Comment:
I don't follow this code well, but do we need multiple rows for EMAIL_TYPE+NOTIFICATION_PERMANENT FAILURE?

Reply:
Ah, good catch. I meant to go back and look at that. One of those was updated from technical failure. I'm checking now if I can just combine those so they aren't weirdly separated out like that.

Reply:
It looks like I can just combine those into one line. I've done so and just pushed the code.